src/swaps/fastbtc/FastbtcSwapProvider.js

```diff
@@ -0,0 +1,254 @@
+import { io } from 'socket.io-client'
+import BN from 'bignumber.js'
+import { mapValues } from 'lodash-es'
+import { SwapProvider } from '../SwapProvider'
+import { v4 as uuidv4 } from 'uuid'
+import { chains, currencyToUnit, unitToCurrency } from '@liquality/cryptoassets'
+import { withInterval } from '../../store/actions/performNextAction/utils'
+import { prettyBalance } from '../../utils/coinFormatter'
+import cryptoassets from '@/utils/cryptoassets'
+import { getTxFee } from '../../utils/fees'
+
+const fastBtcSatoshiFee = 5000
+const fastBtcPercentageFee = 0.2
+
+class FastbtcSwapProvider extends SwapProvider {
+  constructor (config) {
+    super(config)
+    this.socketConnection = io(this.config.agent, {
+      reconnectionDelayMax: 10000
+    })
+
+    this.socketConnection.on('connect', function () {
+      console.log('FatBtc socket connected')
+    })
+
+    this.socketConnection.on('disconnect', function () {
+      console.log('FastBtc socket disconnected')
+    })
+  }
+
+  async getSupportedPairs () {
+    return []
+  }
+
+  async _getHistory (web3Addr) {
+    return new Promise((resolve, reject) => {
+      this.socketConnection.emit('getDepositHistory', web3Addr, (res) => {
+        if (res && res.error) {
+          reject(res.err)
+        }
+        resolve(res)
+      })
+    })
+  }
+
+  async _getAddress (web3Addr) {
+    return new Promise((resolve, reject) => {
+      this.socketConnection.emit('getDepositAddress', web3Addr, (err, res) => {
+        if (err) {
+          reject(err)
+        }
+        resolve(res)
+      })
+    })
+  }
+
+  async _getTxAmount () {
+    return new Promise((resolve, reject) => {
+      this.socketConnection.emit('txAmount', (res) => {
+        resolve(res)
+      })
+    })
+  }
+
+  async getQuote ({ network, from, to, amount }) {
+    const fromAmountInUnit = BN(currencyToUnit(cryptoassets[from], BN(amount)))
+    const validAmountRange = await this._getTxAmount()
+    const isQuoteAmountInTheRange = amount <= validAmountRange.max && amount >= validAmountRange.min
+    if (from !== 'BTC' || to !== 'RBTC' || !isQuoteAmountInTheRange) return null
+    const toAmountInUnit = BN(currencyToUnit(cryptoassets[to],
+      BN(amount).minus(unitToCurrency(cryptoassets[from], fastBtcSatoshiFee))))
+      .times(1 - fastBtcPercentageFee / 100)
+    return {
+      from,
+      to,
+      // TODO: Amounts should be in BigNumber to prevent loss of precision
+      fromAmount: fromAmountInUnit.toNumber(),
+      toAmount: toAmountInUnit.toNumber()
+    }
+  }
+
+  async sendSwap ({ network, walletId, quote }) {
+    if (quote.from !== 'BTC' || quote.to !== 'RBTC') return null
+    const toChain = cryptoassets[quote.to].chain
+    const account = this.getAccount(quote.fromAccountId)
+    const client = this.getClient(network, walletId, quote.from, account?.type)
```

Comment:
I don't think this is like this anymore? I think that you need to pass the accountid not type. Try to get latest from dev to see